lib/stream.js

```diff
@@ -83,19 +102,18 @@ const resumeStream = async stream => {
 
 const applyEncoding = (contents, encoding) => encoding === 'buffer' ? new Uint8Array(contents) : contents;
 
+// Transforms replace `childProcess.std*`, which means they are not exposed to users.
+// However, we still want to wait for their completion.
+const waitForOriginalStreams = (originalStreams, spawned, waitForStream) => originalStreams
+	.filter((stream, index) => stream !== spawned.stdio[index])
+	.map(stream => waitForStream(stream));
+
 // Some `stdin`/`stdout`/`stderr` options create a stream, e.g. when passing a file path.
 // The `.pipe()` method automatically ends that stream when `childProcess` ends.
 // This makes sure we wait for the completion of those streams, in order to catch any error.
-const waitForCustomStreamsEnd = stdioStreamsGroups => stdioStreamsGroups.flat()
+const waitForCustomStreamsEnd = (stdioStreamsGroups, waitForStream) => stdioStreamsGroups.flat()
 	.filter(({type, value}) => type !== 'native' && !isStandardStream(value))
-	.map(({value}) => finished(value, {cleanup: true}));
-
-const throwIfStreamError = stream => stream === null ? [] : [throwOnStreamError(stream)];
-
-const throwOnStreamError = async stream => {
-	const [error] = await once(stream, 'error');
-	throw error;
-};
```

Comment:
Previously we only listened to the `error` event for `childProcess.stdin`. 
Now we wait for its completion, which automatically happens when the child process exits.